vsphere/check.py

```diff
@@ -1186,12 +1188,17 @@ def get_topologyitems_sync(self, instance):
         content = server_instance.RetrieveContent()
         domain = instance["host"]  # candidate also name
 
-        vms = self._vsphere_vms(content, domain)
-        hosts = self._vsphere_hosts(content, domain)
-        datacenters = self._vsphere_datacenters(content, domain)
-        datastores = self._vsphere_datastores(content, domain)
-        clustercomputeresources = self._vsphere_clustercomputeresources(content, domain)
-        computeresource = self._vsphere_computeresources(content, domain)
+        regexes = {
+            'host_include': instance.get('host_include_only_regex'),
+            'vm_include': instance.get('vm_include_only_regex')
+        }
+
+        vms = self._vsphere_vms(content, domain, regexes)
+        hosts = self._vsphere_hosts(content, domain, regexes)
+        datacenters = self._vsphere_datacenters(content, domain, regexes)
```

Comment:
Looking at definition of  _is_excluded - the only supported entity types are hosts and virtual machines.
Introducing regexes parameters to datacenters, datastores, clustercomputeresources, computeresource is not needed.

```
    def _is_excluded(obj, regexes, include_only_marked):
        """
        Return `True` if the given host or virtual machine is excluded by the user configuration,
        i.e. violates any of the following rules:
        * Do not match the corresponding `*_include_only` regular expressions
        * Is "non-labeled" while `include_only_marked` is enabled (virtual machine only)
```

Reply:
I passed it because we used hosts and VMs in other resources and then create relations as well. So we tried to avoid to get the relations for filtered VMs and hosts. Let me know if it is clear.

Reply:
@vishalxebia That does not mean, that you should copy paste that everywhere.

Example:  _vsphere_datacenters  - no references to  _is_excluded, parameters not needed.

Reply:
yeah correct.

Reply:
removed the parameter from `_vsphere_datacenters`.